pkg/segment/utils/number_test.go

```diff
@@ -0,0 +1,396 @@
+package utils
+
+import (
+	"math"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+// Test_isInvalid tests if the number is invalid. It should return invalidType.
+func Test_isInvalid(t *testing.T) {
+	n := Number{}
+	n.SetInvalidType()
+	assert.Equal(t, n.ntype(), invalidType)
+}
+
+// Test_invalidTypeReturnsErrorForInt64AndFloat64 tests converting invalidType to Int64 and Float64.
+func Test_invalidTypeReturnsErrorForInt64AndFloat64(t *testing.T) {
+	n := Number{}
+	n.SetInvalidType()
+
+	_, err := n.Int64()
+	assert.NotEqual(t, nil, err)
```

Comment:
nit: you can do: `assert.NotNil(t, err, "custom informational message")`. 

Same comment elsewhere

Reply:
The NotNil gives more information. Makes sense. Changed it